app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -10,6 +11,9 @@ def resource_locks
 
       def plan(job_invocation, host, template_invocation, proxy, connection_options = {})
         action_subject(host, :job_name => job_invocation.job_name)
+        link!(job_invocation)
+        link!(template_invocation)
+
```

Comment:
We need to do that as soon as possible, before any other error occurs, so that one can get to the errors form the linked resource